src/main/java/org/opentestsystem/ap/ims/service/FileUploadService.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright 2017 Regents of the University of California.
+ *
+ *  Licensed under the Educational Community License, Version 2.0 (the "license");
+ *  you may not use this file except in compliance with the License. You may
+ *  obtain a copy of the license at
+ *
+ *  https://opensource.org/licenses/ECL-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.opentestsystem.ap.ims.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.opentestsystem.ap.common.model.Attachment;
+import org.opentestsystem.ap.common.model.ItemBankUser;
+import org.opentestsystem.ap.common.repository.ItemRepository;
+import org.opentestsystem.ap.common.util.SecurityUtil;
+import org.opentestsystem.ap.ims.model.BrailleFile;
+import org.opentestsystem.ap.ims.model.UploadedFile;
+import org.opentestsystem.ap.ims.util.BrailleFileUtil;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * Handles file requests like uploading, downloading, and retrieving item files.
+ */
+@Slf4j
+@Component
+public class FileUploadService {
+
+    private final BrailleFileUtil brailleFileUtil;
+
+    private final ItemRepository itemRepository;
+
+    private final SecurityUtil securityUtil;
+
+    public FileUploadService(final BrailleFileUtil brailleFileUtil,
```

Comment:
The intention is one service for all file uploads and possibly multiple utilities for the different upload types.  Maybe that is unnecessary and there could be one FileUtil.  I could even see extracting the validation out of the utility.  I wasn't sure I liked it there but decided to keep all things braille file in that one util.  Maybe there is a FileUtil and a FileValidator class that server all file types.  That seems better.  I'll look to separate this as part of the tech clean up.  I'll put it on the ticket now.